### PR TITLE
Avoid heap corruption issue by setting the property via the DP instead of the property setter

### PR DESCRIPTION
**Pull request by StephenLPeters**
An OS bug was fixed in 20h1 that resolved the heap corruption bug our code is hitting on 19h1 and 19h2. Fixes https://github.com/microsoft/microsoft-ui-xaml/issues/5435

### CODE REVIEW

File: dev/inc/SharedHelpers.h [hunk 1]
```diff
@@ -63,6 +64,7 @@ class SharedHelpers
 
     // Actual OS version checks
     static bool IsAPIContractV14Available(); // 21H1
+    static bool IsAPIContractV10Available();  // 20H1
```
Comment (asklar) [inline dev/inc/SharedHelpers.h:67]:
```suggestion
    static bool IsAPIContractV10Available(); // 20H1
```

File: dev/dll/XamlControlsResources.cpp [hunk 2]
```diff
@@ -204,47 +204,61 @@ void XamlControlsResources::UpdateAcrylicBrushesDarkTheme(const winrt::IInspecta
...
+{
+    if (SharedHelpers::Is20H1OrHigher())
+    {
+        brush.TintLuminosityOpacity(luminosityValue);
+    }
+    else
+    {
+        // On 19h1 and 19h2 there is a heap corruption bug that occurs when setting the TintLuminosityOpacity via the property,
+        // So we'll set it by the DP instead.
+        brush.SetValue(winrt::Windows::UI::Xaml::Media::AcrylicBrush::TintLuminosityOpacityProperty(), box_value(luminosityValue));
```
Comment (asklar) [inline dev/dll/XamlControlsResources.cpp:258]: IIRC the only difference is that this doesn't trigger the "notify on property change" handlers, which is probably ok for this prop but would be worth adding a comment for

**Review by asklar — APPROVED** [review]

**Review by miniksa — APPROVED** [review]
This matches my understanding of the issue and the proposal that I had made in the internal email investigation thread we had. Thank you for doing this! Hopefully we can get a release with this fix very soon!

**Pull request merged**